src/main/java/com/artozersky/HackerNewsAPI/cache/CacheEntityService.java

```diff
@@ -0,0 +1,60 @@
+package com.artozersky.HackerNewsAPI.cache;
+
+import java.util.List;
+import com.artozersky.HackerNewsAPI.model.impl.NewsPostModelImpl;
+
+/**
+ * Interface for cache operations related to {@link NewsPostModelImpl} entities.
+ * Provides methods for retrieving, storing, and managing cached posts.
+ */
+public interface CacheEntityService {
+
+    /**
+     * Retrieves a post from the cache by its ID.
+     *
+     * @param postId The ID of the post to retrieve.
+     * @return The cached {@link NewsPostModelImpl} if found, or null if not present in the cache.
+     */
+    NewsPostModelImpl getPostFromCacheById(Long postId);
```

Comment:
I think that from cache is not a good name the object you use is called cache so it is pretty understandable without the from cache name. change it in all file to be without from cache.

Reply:
fixed